From fcbdcc9b571d4382fc8e84665b1293cfb66acfea Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mika=20Bostr=C3=B6m?= <mika.bostrom@nomovok.com>
Date: Fri, 20 Apr 2012 14:33:42 +0300
Subject: [PATCH 6/6] Install generated Qt3DQuick headers

The classname headers are generated during build but not installed.
Fix the oversight.

Add a new file to include the generated headers. (This is identical to
modifications done to src/threed/ path.)
---
 src/quick3d/genheaders.pri |    7 +++++++
 src/quick3d/quick3d.pro    |    7 +++++++
 2 files changed, 14 insertions(+), 0 deletions(-)
 create mode 100644 src/quick3d/genheaders.pri

diff --git a/src/quick3d/genheaders.pri b/src/quick3d/genheaders.pri
new file mode 100644
index 0000000..09939ae
--- /dev/null
+++ b/src/quick3d/genheaders.pri
@@ -0,0 +1,7 @@
+GENERATED_HEADERS = \
+    ../../include/Qt3DQuick \
+    ../../include/Qt3DQuick/QQuickViewport \
+    ../../include/Qt3DQuick/QQuickEffect \
+    ../../include/Qt3DQuick/QQuickAnimation3D \
+    ../../include/Qt3DQuick/QQuickMesh \
+    ../../include/Qt3DQuick/QQuickItem3D
diff --git a/src/quick3d/quick3d.pro b/src/quick3d/quick3d.pro
index d48b88b..54f6fd6 100644
--- a/src/quick3d/quick3d.pro
+++ b/src/quick3d/quick3d.pro
@@ -44,6 +44,13 @@ devel.path = $$[QT_INSTALL_HEADERS]/$$TARGET
 
 INSTALLS += devel
 
+# Automatically generated classname headers are not installed
+# by default rules. genheaders.pri contains a list of
+# the generated files
+include(genheaders.pri)
+genheaders.files = $$GENERATED_HEADERS
+genheaders.path = $$[QT_INSTALL_HEADERS]/$$TARGET
+INSTALLS += genheaders
 
 
 
-- 
1.7.5.4

